### ISSUE TITLE
I got a error message when I ran binary after build.

### ISSUE THREAD

**Issue by systembug** [copperspice/kitchensink#4]
OS: Ubuntu 18.04
CopperSpice Version: 1.6
KitchenSink Version: 1.6

I got a error message as below:

The application failed to start because the platform plugin was not found or did not load.
Requested Plugin Key: "xcb"

Reinstalling the application may resolve this problem.

Could I solve this problem?

**Comment by bgeller** [comment]
Happy to help out.  

Did you build CopperSpice from source as well as KitchenSink? 
Can you first check your CopperSpice install/lib folder and verify that you have **CsGuiXcb1.6.so**. 
When you run the CMake deploy for KitchenSink, this file should be copied to deploy/platforms.

We retest some of the pre-built binary KS files for Unix and the worked as expected.

**Comment by vdeurzen** [comment]
Hello @systembug, I've had a similar issue before and managed to solve it. An important step to finalize the build process is to run `make install` or `ninja install` (depending on your chosen configuration). The installation step will ensure that all necessary plugins are put in the expected places.

I've started a blog series about CopperSpice that explains these steps in more detail, you can find it here: [https://eolas.engineering/tags/copperspice/](https://eolas.engineering/tags/copperspice/)

**Comment by systembug** [comment]
> Happy to help out.
> 
> Did you build CopperSpice from source as well as KitchenSink?
> Can you first check your CopperSpice install/lib folder and verify that you have **CsGuiXcb1.6.so**.
> When you run the CMake deploy for KitchenSink, this file should be copied to deploy/platforms.
> 
> We retest some of the pre-built binary KS files for Unix and the worked as expected.

Thanks for reply.
Yes, I already checked for **CsGuiXcb1.6.so.** This file exists in the lib folder.
When I had built KitchenSink 1.6, I hadn't gotten any errors.
Can I debug CopperSpice when I run KitchenSink 1.6?
As far as I know, QT supports to the debug mode with options.

**Comment by systembug** [comment]
> Hello @systembug, I've had a similar issue before and managed to solve it. An important step to finalize the build process is to run `make install` or `ninja install` (depending on your chosen configuration). The installation step will ensure that all necessary plugins are put in the expected places.
> 
> I've started a blog series about CopperSpice that explains these steps in more detail, you can find it here: https://eolas.engineering/tags/copperspice/

Thank you for your comments.
I read your article. But I think build options didn't contain multimedia on yours.
This way can avoid this problem but cannot solve it.

**Comment by bgeller** [comment]
> Yes, I already checked for CsGuiXcb1.6.so. This file exists in the lib folder.

This is good to know, you have the platform file built.  

The next question is, did you install KitchenSink?  If you did then you will have a folder called "deploy" with the executable. All of the CS so files should be in the same folder.  There will also be a "deploy/platforms" folder and the XCB plugin will be there.  If any of these files are missing then you did not install KS after you built the project.

Barbara

**Comment by systembug** [comment]
> > Yes, I already checked for CsGuiXcb1.6.so. This file exists in the lib folder.
> 
> This is good to know, you have the platform file built.
> 
> The next question is, did you install KitchenSink? If you did then you will have a folder called "deploy" with the executable. All of the CS so files should be in the same folder. There will also be a "deploy/platforms" folder and the XCB plugin will be there. If any of these files are missing then you did not install KS after you built the project.
> 
> Barbara

Good answer. It works!
Thank you for your help.

**Comment by stefanos82** [comment]
Those who just want to build the program and install it locally, without the need to pollute their `/usr/local/` like I accidentally did and nearly had a heart attack :smile: , please run the following command inside the `build` folder you just created: 

```sh
cmake .. -G "Ninja" 
    -DCMAKE_BUILD_TYPE=Release 
    -DCMAKE_INSTALL_PREFIX:PATH="../output"
    -DCMAKE_PREFIX_PATH="/c/cs_lib/" .
```
hint: please change `CMAKE_PREFIX_PATH` accordingly.

Then run `ninja && ninja install` and that should be it; the first command builds the application and the latter installs it in project's root directory under the `output` sub-directory.

In other words, the tree should directory should look something like this:

```
coppersink
    > build
    > output
    > forms
    > resources
    > etc...
```

**Comment by bgeller** [comment]
@stefanos82 

I would like to clarify some of your notations.  Not sure what "coppersink" is but I have a feeling you meant "KitchenSink" which is our demo program.

(A)  There are directions for building **CopperSpice** in our "CS Overview" documentation and we do recommend adding a PREFIX path. We also recommend you build CS _out of tree_. Our suggestion is to have a unique path for source, build, and install. 

Refer to this link and then scroll down to "Build Process"
https://www.copperspice.com/docs/cs_overview/build-overview.html


(B)   When you are building **KitchenSink** you should specify the PREFIX.  

Refer to this link 
https://www.copperspice.com/docs/cs_overview/demo-ks.html#ks-build-script

**Comment by stefanos82** [comment]
> @stefanos82
> 
> I would like to clarify some of your notations. Not sure what "coppersink" is but I have a feeling you meant "KitchenSink" which is our demo program.
> 

Classic me I invented my own word again lol; yes, I meant `kitchensink`.

> (A) There are directions for building **CopperSpice** in our "CS Overview" documentation and we do recommend adding a PREFIX path. We also recommend you build CS _out of tree_. Our suggestion is to have a unique path for source, build, and install.
> 
> Refer to this link and then scroll down to "Build Process"
> https://www.copperspice.com/docs/cs_overview/build-overview.html
>

I have managed to compile CS without much trouble. I do though have had problem to follow the documentation as I got lost a couple of times; for instance, the example with `/c/cs_*` format.

A more *obvious* example with cliche names would help a lot more, at least my case which I'm a visual person.

> (B) When you are building **KitchenSink** you should specify the PREFIX.
> 
> Refer to this link
> https://www.copperspice.com/docs/cs_overview/demo-ks.html#ks-build-script

Don't ask me why, but I could not see clearer what I'm reading right now; the example is nearly identical with my code except the use of `:PATH` appended at the end of `-DCMAKE_INSTALL_PREFIX`.

I guess I should have read the docs multiple times first and then attempt to compile the code.
